ingest/src/main/java/com/microsoft/azure/kusto/ingest/StreamingIngestClient.java

```diff
@@ -130,12 +131,12 @@ public IngestionResult ingestFromStream(StreamSourceInfo streamSourceInfo, Inges
 
         } catch (DataClientException | IOException e) {
             log.error(e.getMessage(), e);
-            if (e.getCause() instanceof DataWebException && "Error in post request".equals(e.getMessage())) {
-                validateEndpointServiceType(connectionDataSource, EXPECTED_SERVICE_TYPE);
-            }
             throw new IngestionClientException(e.getMessage(), e);
         } catch (DataServiceException e) {
             log.error(e.getMessage(), e);
+            if (e.getMessage() != null && e.getMessage().contains("404")) {
```

Comment:
Better to match more than just 404. Maybe use a regex to ensure \s\d\d\d\s or similar.

Reply:
Can we relay on more that?

Reply:
More would be ideal - it requires testing to see the range of possibilities, But I suspect a text match of "404" is insufficient.

Reply:
Ok, I fixed it properly.
We now transfer the actual response object, and then we can see if it's actually 404 LMK what you think